tox.ini

```diff
@@ -1,40 +1,40 @@
 [tox]
 envlist =
-    py27,py35,py36,py37,pypy,pypy3,docs,coverage
+    py27,py35,py36,py37,py38,pypy,pypy3,coverage,docs
+# Note: if you add new Python versions, please add them to
+# [testenv:coverage] depends as well!
 
 [testenv]
 commands =
-    zope-testrunner --test-path=src {posargs:-vc}
-# without explicit deps, setup.py test will download a bunch of eggs into $PWD
+    coverage run -p -m zope.testrunner --test-path=src []
+extras =
+    test
 deps =
-    .[test]
-
-[testenv:pypy3]
+    coverage
+setenv =
+     ZOPE_INTERFACE_STRICT_IRO=1
 # On certain platforms, at least OS X, pypy3 does not correctly read
 # a LANG environment variable; when cpython3.2 would return 'utf-8'
 # from locale.getpreferredencoding(), pypy3 returns 'us-ascii', causing
 # two test failures. Fix that by setting something it does read.
-setenv =
-    LC_CTYPE=en_US.UTF-8
+     pypy3: LC_CTYPE=en_US.UTF-8
 
 [testenv:coverage]
-usedevelop = true
-basepython =
-    python3.6
+skip_install = true
 commands =
-    coverage run -m zope.testrunner --test-path=src {posargs:-vc}
-    coverage report --fail-under=92
-deps =
-    {[testenv]deps}
-    coverage
+    coverage erase
+    coverage combine
+    coverage html -i
```

Comment:
Why is `-i` ("Ignore errors while reading source files") needed here?

I notice that `.coveragerc` here doesn't have the

```
[paths]
source =
    src/
    .tox/*/lib/python*/site-packages/
    .tox/pypy*/site-packages/
```

that we use [elsewhere](https://github.com/zopefoundation/zope.annotation/blob/503f3fc0fede04c97a02576a013401751a43d651/.coveragerc#L8-L12) to make `coverage combine` work correctly when `[testenv]` doesn't use `usedevelop = true`.  (It is in fact my preferred solution for making coverage find the right source files, because usedevelop = false allows us to catch certain packaging problems sometimes.)

With that added, would you still need the `-i` here?

Reply:
My use for -i is about [temporary files left in an editor](https://github.com/zopefoundation/zope.traversing/pull/15#discussion_r400308663), so I would still prefer the '-i'.